### PR TITLE
Eliminate bundle audit vulnerabilities that are likely not introduced by the commit.

### PR DESCRIPTION
**Pull request by ghbren**
Code stores the gem name without version in snippet.  Then looks for the snippet in git diff.
If there are multiple versions of the same gem, then it may be an issue. However, this PR is just here to eliminate vulnerabilities that are not likely introduced commit.

PR assumes Gemfile.lock was auto-created.  If the user manually updates it to have some unconventional format, then things may break.

PR has also been tested using docker build ..., docker run local salus with `--sarif_full_dif ... --git-diff ...` to make sure
1.  vuls caused by changed lines do not show up in the diffed results.
2. vuls introduced by the commit still show up.

### CODE REVIEW

File: lib/sarif/bundle_audit_sarif.rb [hunk 1]
```diff
@@ -50,5 +51,12 @@ def sarif_level(severity)
         SARIF_WARNINGS[:error]
       end
     end
+
+    def self.snippet_possibly_in_git_diff?(snippet, lines_added)
```
Comment (joshua-ostrom) [inline lib/sarif/bundle_audit_sarif.rb:55]: Is the only used by specs? If so can we make this an rspe helper?

File: lib/sarif/bundle_audit_sarif.rb [hunk 2]
```diff
@@ -29,6 +29,7 @@ def parse_issue(issue)
       if issue[:line_number]
         result[:start_line] = issue[:line_number]
         result[:start_column] = 1
+        result[:code] = issue[:name].to_s
```
Comment (joshua-ostrom) [inline lib/sarif/bundle_audit_sarif.rb:32]: Can you include the comment present in your PR description here in the code for readability?
# Code stores the gem name without version in snippet.

File: lib/sarif/bundle_audit_sarif.rb (hunk 1 above)
Comment (ghbren) [reply lib/sarif/bundle_audit_sarif.rb:55]: No. This code is used outside specs.
In fact, `docker run ... --git-diff` calls this method.

File: lib/sarif/bundle_audit_sarif.rb (hunk 2 above)
Comment (ghbren) [reply lib/sarif/bundle_audit_sarif.rb]: added, great suggestion!

**Review by joshua-ostrom — APPROVED** [review]

File: spec/lib/sarif/bundle_audit_spec.rb [hunk 3]
```diff
@@ -135,4 +138,30 @@
...
+        snippet = 'helloworld'
+        git_diff = File.read(git_diff_file)
+        new_lines_in_git_diff = Sarif::BaseSarif.new_lines_in_git_diff(git_diff)
+        r = Sarif::BundleAuditSarif.snippet_possibly_in_git_diff?(snippet, new_lines_in_git_diff)
+        expect(r).to be true
+        snippet = 'bye'
+        r = Sarif::BundleAuditSarif.snippet_possibly_in_git_diff?(snippet, new_lines_in_git_diff)
+        expect(r).to be true
+
+        # should not match part of the package name
```
Comment (saikat056) [inline spec/lib/sarif/bundle_audit_spec.rb:155]: Can we write this part as a separate unit test?

```
it 'should not match part of the package name' do
  ...
end
```

File: spec/lib/sarif/bundle_audit_spec.rb [hunk 4]
```diff
@@ -135,4 +138,30 @@
...
+        snippet = 'hello'
+        r = Sarif::BundleAuditSarif.snippet_possibly_in_git_diff?(snippet, new_lines_in_git_diff)
+        expect(r).to be false
+
+        # should not match package that was in git diff but not added with this commit
```
Comment (saikat056) [inline spec/lib/sarif/bundle_audit_spec.rb:160]: Please write this as a separate unit test

it 'should not match package that was in git diff but not added with this commit' do
end

**Review by saikat056 — APPROVED** [review]

**Pull request merged**